PCF - 250 Results: Add implementation for switching between comparisons Part 2

esanuandra opened the pull request:
This PR fulfills [Results: Add implementation for switching between comparisons](https://mozilla-hub.atlassian.net/browse/PCF-250)
subtask: [Populate all revisions dropdown with real data](https://mozilla-hub.atlassian.net/browse/PCF-282)

STR:

1. Go to http://localhost:3000/ or https://deploy-preview-516--mozilla-perfcompare.netlify.app/
2. Select some base and new revisions for comparison
3. Click `Compare` to be redirected to results page
4. Click on `All revisions` dropdown
5. The revisions that were selected on the home page should be in this dropdown

![Screenshot 2023-08-17 at 5 27 44 PM](https://github.com/mozilla/perfcompare/assets/63001299/9350c04a-e501-4288-8e2f-054eb3e6516d)

Carla-Moz reviewed (commented):
LGTM so far! Also, could you please had a screenshot next to your PRs or some description.

esanuandra commented:
> LGTM so far! Also, could you please had a screenshot next time to your PRs or some description.

@Carla-Moz PR is ready for review now, it was WIP before. I updated the description and added a screenshot.

julienw reviewed (approved):
I didn't try it but this looks good from a quick overview.

Please look at extracting the selector soon :-)

esanuandra commented on src/components/CompareResults/beta/RevisionSelect.tsx:
```diff
@@ -34,6 +36,19 @@ function RevisionSelect() {
   const dispatch = useDispatch();
   const { activeComparison } = useAppSelector((state) => state.comparison);
 
+  const newRevisions = useAppSelector((state) => {
+    return state.selectedRevisions.new.map((item) =>
+      truncateHash(item.revision),
+    );
+  });
```
[PCF-290 Create memoized selector for populating all revisions dropdown](https://mozilla-hub.atlassian.net/browse/PCF-290)
